backends/arm/arm_backend.py

```diff
@@ -162,65 +163,82 @@ def vela_compile(tosa_fb):
 
         np_path = os.path.join(tmpdir, "output", "out_sg0_vela.npz")
         blocks = b""
+
         with np.load(np_path, allow_pickle=False) as data:
+            # Construct our modified output_blocks with data in a form easily
+            # digested on the device side
+            bin_blocks = {"vela_bin_stream": b""}
+
+            # copy command data through unmodified
+            bin_blocks["cmd_data"] = data["cmd_data"].tobytes()
+
+            # copy weight data through unmodified
+            bin_blocks["weight_data"] = data["weight_data"].tobytes()
+
+            # Add a block for scratch, inputs and outputs;  scratch shape is a 1 element
+            # array giving us size in bytes so extract this and add a block of 0's.
+            # Currently we preallocated this on the host to provide SRAM for computation.
+            block_length = data["scratch_shape"][0].item()
+            bin_blocks["scratch_data"] = b"\x00" * block_length
+
+            # compose the input related arrays into per-input data to simplify
+            # the runtime code
+            inputs = struct.pack("<i", len(data["input_shape"]))
+            for i in range(len(data["input_shape"])):
+                input_shape = data["input_shape"][i]
+                input_elem_size = data["input_elem_size"][i]
+                input_offset = data["input_offset"][i]
+                input_region = data["input_region"][i]
+                assert len(input_shape) <= 4
+                inp_pad = input_shape.tolist() + [0] * (4 - len(input_shape))
+                input_struct = struct.pack(
+                    "<iiiiiii", *inp_pad, input_elem_size, input_offset, input_region
+                )
```

Comment:
make a function?
`def create_struct(data: Dict[str, Any], type: Enum, index: int) -> Struct:`
type can be an enum of input, output

Reply:
fixed.